behaviors/smart-contracts/system/_Deployments.md

```diff
@@ -81,5 +103,41 @@ To retrieve the code, `getCodeParts()` in combination with `getCodePart()` shoul
 #### Behavior
 * Takes service name as argument.
 * Takes index as argument.
+* Takes version as optional argument (if not present defaults to `0`)
 * Read code from state key `<name>.Code.<counter>` by calling `State.ReadBytesByKey`.
 * Returns error if the contract was not deployed
+
+## `getVersion` (method)
+
+Returns current version of the contract (`0` for a contract deployed with `deployService`).
+
+#### Permissions
+* `External` (caller can be anyone).
+* `ReadOnly` (does not change state).
+
+#### Behavior
+* Takes service name as argument.
+* Read code from state key `<name>.Code.<counter>` by calling `State.ReadUint32ByKey`.
+* Returns `0` if the contract was not deployed
+
+## `allowVersioning` (method)
```

Comment:
is there a sense in separating governance concerns to a separate contract? this way a chain owner could deploy the policy contract so that each chain can be configured with different upgrade policy (ie, upgrade delays, upgrade by voting, upgrade appeals, permissions etc..)

Reply:
Yeah, I think policy contract is a great idea.